templates/todo/api/js/src/app.ts

```diff
@@ -17,7 +17,22 @@ export const createApp = async (): Promise<Express> => {
     await configureMongoose(config.database);
     // Middleware
     app.use(express.json());
-    app.use(cors());
+
+    const apiUrl:string|undefined = process.env.REACT_APP_WEB_BASE_URL;
+    if ( typeof apiUrl !== "undefined"){
+        app.use(cors({
+            origin: ["https://portal.azure.com",
+                "https://ms.portal.azure.com",
+                "http://localhost:3000/",
+                apiUrl]
+        }));
+        // __filename caused fails when interact with azure portal. hardcode file name here.
+        console.log("CORS with "+origin[2]+" is allowed for local host debugging. If you want to change pin number, go to app.ts.");
```

Comment:
`origin` does not exists.  It is created inside the call to `cors()` only.  If you want to use it here, you need to do something like:

const origin = ["foo", "bar"]
app.use(cors({
            origin: origin));
        // __filename caused fails when interact with azure portal. hardcode file name here.
        console.log("CORS with "+origin[2]+" is allowed for local host debugging. If you want to change pin number, go to app.ts.");